src/main/java/com/autotune/analyzer/recommendations/updater/vpa/VpaUpdaterImpl.java

```diff
@@ -0,0 +1,377 @@
+     */
+    @Override
+    public boolean isUpdaterInstalled() {
+        try {
+            LOGGER.info(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.CHECKING_IF_UPDATER_INSTALLED,
+                    AnalyzerConstants.RecommendationUpdaterConstants.SupportedUpdaters.VPA);
+            // checking if VPA CRD is present or not
+            boolean isVpaInstalled = false;
+            CustomResourceDefinitionList crdList = apiextensionsClient.v1().customResourceDefinitions().list();
+            if (null != crdList && null != crdList.getItems() && !crdList.getItems().isEmpty()) {
+                isVpaInstalled = crdList.getItems().stream().anyMatch(crd -> AnalyzerConstants.RecommendationUpdaterConstants.VPA.VPA_PLURAL.equalsIgnoreCase(crd.getSpec().getNames().getKind()));
+            }
+            if (isVpaInstalled) {
+                LOGGER.info(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.FOUND_UPDATER_INSTALLED, AnalyzerConstants.RecommendationUpdaterConstants.SupportedUpdaters.VPA);
+            } else {
+                LOGGER.error(AnalyzerErrorConstants.RecommendationUpdaterErrors.UPDATER_NOT_INSTALLED);
+            }
+            return isVpaInstalled;
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Checks if a Vertical Pod Autoscaler (VPA) object with the specified name is present.
+     *
+     * @param vpaName String containing the name of the VPA object to search for
+     * @return true if the VPA object with the specified name is present, false otherwise
+     */
+    private boolean checkIfVpaIsPresent(String vpaName) {
+        try {
+            if (null == vpaName || vpaName.isEmpty()) {
+                throw new Exception(AnalyzerErrorConstants.RecommendationUpdaterErrors.INVALID_VPA_NAME);
+            } else {
+                LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.CHECKING_IF_VPA_PRESENT, vpaName));
+                NamespacedVerticalPodAutoscalerClient client = new DefaultVerticalPodAutoscalerClient();
+                VerticalPodAutoscalerList vpas = client.v1().verticalpodautoscalers().inAnyNamespace().list();
+
+                if (null != vpas && null != vpas.getItems() && !vpas.getItems().isEmpty()) {
+                    // TODO:// later we can also check here is the recommender is Kruize to confirm
+                    for (VerticalPodAutoscaler vpa : vpas.getItems()) {
+                        if (vpaName.equals(vpa.getMetadata().getName())) {
+                            LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.VPA_WITH_NAME_FOUND, vpaName));
+                            return true;
+                        }
+                    }
+                }
+                LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.VPA_WITH_NAME_NOT_FOUND, vpaName));
```

Comment:
This should be LOGGER.error right? Others should be debug

Reply:
Updated this one to `error` message.